### PR TITLE
Debian based images

### LINKED ISSUE #50: Consider switching to Debian as base image
Currently, we are using [Alpine](https://wiki.alpinelinux.org/wiki/Docker) as the base image for almost all services. At the time, this was the recommended base image and [Docker themself move their own images on Alpine](https://brianchristner.io/docker-is-moving-to-alpine-linux/) too.

2020 could be the year we switch back to Debian as nowadays they provide official image that no so big (like `debian:buster-slim`?). Here is some pro/cons:

Pros:
- a dev env more similar to classical prod env
- easier to use latest PHP version (almost impossible with Alpine)
- Alpine uses the `musl` library instead of classical libc, which used to break some tool like Blackfire

Cons:
- base image a bit bigger (~5Mo VS 60Mo)
- more different base images among all services thus more images to build and store
- bigger and different base images also slow CI build and can cost more credits (but should be less problematic than when we used normal Debian images weighting 500Mo)

Some read:
- https://phusion.github.io/baseimage-docker/
- https://github.com/docker-slim/docker-slim

### PR DESCRIPTION
**Pull request by lyrixx**
fixes #50 
## Bench

I ran the following command for all benchs.
But for some benchs, I also delete some docker objects.

```
fab destroy && fab start
```

### Time with base image downloaded

alpine: 0m56.873s

debian 2m22.495s

### Initial image size

alpine: 5.55MB

debian: 69.2MB

### Final image size

alpine:
```
app_php-base      59.8MB
app_frontend      65.8MB
app_builder       324MB
```

debian:
```
app_php-base      174MB
app_frontend      208MB
app_builder       335MB
```

### Runtime performance

I ran `ab -c 100 -n 1000 https://app.test/` against a new symfony application in **dev** with the default controller created by the maker bundle

alpine

```
Concurrency Level:      100
Time taken for tests:   8.613 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      51373000 bytes
HTML transferred:       51104000 bytes
Requests per second:    116.11 [#/sec] (mean)
Time per request:       861.284 [ms] (mean)
Time per request:       8.613 [ms] (mean, across all concurrent requests)
Transfer rate:          5824.90 [Kbytes/sec] received
```

debian

```
Concurrency Level:      100
Time taken for tests:   5.908 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      51373000 bytes
HTML transferred:       51104000 bytes
Requests per second:    169.25 [#/sec] (mean)
Time per request:       590.837 [ms] (mean)
Time per request:       5.908 [ms] (mean, across all concurrent requests)
Transfer rate:          8491.17 [Kbytes/sec] received
```

### Pros

* It's simpler (we know better debian than alpine)
* Better DX (sudo for example)
* PHP 7.4 (VS PHP 7.3)

### Cons

* base image is header (+65mb)
* final images are heavier (+150mb for FPM, +10Mb for the builder)
* it's slower to build from scratch (+1m30s)

---

So, what we should do ?
We are going a new major release with a big BC break: [python 3 + invoke instead of fabric](https://github.com/jolicode/docker-starter/pull/61) so we have a window to break everything

* :tada: for sticking on bare alpine
* :rocket: to switch to debian

### CODE REVIEW

**Comment by Nek-** [comment]
@lyrixx why not using the official php (alpine) image as base?

**Comment by lyrixx** [comment]
It's all explained in the issue.

**Comment by Nek-** [comment]
@lyrixx you were using the [official alpine image](https://hub.docker.com/_/alpine). Which is completely different from the [php official image](https://hub.docker.com/_/php).

**Comment by lyrixx** [comment]
> Because correct me if I'm wrong but... The following image seems cool enough for your base:

again, we have already talked about that in the issue. Here is the direct link to it: https://github.com/jolicode/docker-starter/issues/50#issuecomment-574599570


> Also I'm not sure to understand why you want sudo while you can do `docker-compose exec --user=root`.

I's more convenient when you need to do something to run `sudo vim /etc/XXX`, than `CTRL+D` + your command. It's faster :) But sudo is juste an example. More tools are available and it's more standard

**Comment by jbdelhommeau** [comment]
Do you bench performance diff between debian and alpine ?

**Comment by lyrixx** [comment]
You are talking about runtime?

**Comment by bastnic** [comment]
We also ran on an obsure issue with `GLOB_BRACE` not available on alpine that destroyed our DX for months by reducing php perf by half (see issue https://github.com/symfony/symfony/issues/35009).

This is fixed but we do not have any insurance to not have any other difference and issues specific to our alpine dev base env. As we does not have any alpine in production, we prefer deb-based distro.

**Comment by lyrixx** [comment]
I'm more and more in favor of Debian ATM. And according to the vote, it looks like other people too (some people voted for both options :man-facepalming::skin-tone-3: though)

**Comment by jbdelhommeau** [comment]
> You are talking about runtime?

Check for differences in the symonfy initialisation, or another global runtime performance (nginx, php fpm, etc)

**Comment by lyrixx** [comment]
OK, I have done some runtime bench. See the "Runtime performance" section in the PR body.

TL;DR: 

alpine: Requests per second:    116.11 [#/sec] (mean)
debian: Requests per second:    169.25 [#/sec] (mean)

**warning**: alpine used PHP 7.3 and debian is using PHP 7.4
